app/src/componentLibrary/datePicker/header/datePickerHeader.jsx

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2023 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Parser from 'html-react-parser';
+import classNames from 'classnames/bind';
+import { useIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import { useMemo } from 'react';
+import { Dropdown } from 'componentLibrary/dropdown';
+import { COMMON_LOCALE_KEYS, months } from 'common/constants/localization';
+import styles from './datePickerHeader.scss';
+import DatePickerHeaderIcon from './img/calendar-arrow-inline.svg';
+
+const cx = classNames.bind(styles);
+
+const messagesMonths = Object.fromEntries(
+  Object.entries(COMMON_LOCALE_KEYS).filter(([key]) => months.includes(key)),
+);
```

Comment:
Can you extract the months simpler?
F.e. like here

![image](https://user-images.githubusercontent.com/31861805/217198462-456f6511-bbcc-4714-a140-99f74afef00c.png)